### PR TITLE
fix(runtime): Properly decRef record and array values

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

**Review by phated — APPROVED** [review]
Excellent! What's the best way to have a test for this?

File: runtime/src/core/memory.js [hunk 1]
```diff
@@ -417,15 +428,33 @@ export class ManagedMemory {
...
+          case GRAIN_ARRAY_HEAP_TAG: {
+            let x = genericHeapValUserPtr / 4;
+            let arity = view[x + 1];
+            trace(`traversing ${arity} array vals on array 0x${this._toHex(userPtr)}`);
+            for (let i = 0; i < arity; ++i) {
+              yield view[x + 2 + i];
+            }
+            break;
+          }
           default:
```
Comment (peblair) [inline runtime/src/core/memory.js:458]: Maybe we should emit some sort of warning here for unknown tags to prevent bugs like this in the future?

**Review by peblair — APPROVED** [review]

**Comment by ospencer** [comment]
@phated The best way to have a test for this is to just add a GC test that verifies that after arrays and records are allocated, their memory is reclaimed. I believe that I've found a bug in the way we currently write GC tests that doesn't allow me to properly write one right now. I'm going to open an issue to address it, though.

**Comment by ospencer** [comment]
https://github.com/grain-lang/grain/issues/504

**Pull request merged**